toolchain/check/context.cpp

```diff
@@ -799,30 +801,79 @@ auto Context::SetBlockArgResultBeforeConstantUse(SemIR::InstId select_id,
   }
 }
 
-auto Context::AddCurrentCodeBlockToFunction(Parse::NodeId node_id) -> void {
-  CARBON_CHECK(!inst_block_stack().empty(), "no current code block");
-
-  if (return_scope_stack().empty()) {
-    CARBON_CHECK(node_id.is_valid(),
-                 "No current function, but node_id not provided");
-    TODO(node_id,
-         "Control flow expressions are currently only supported inside "
-         "functions.");
+auto Context::AddToRegion(SemIR::InstBlockId block_id) -> void {
+  if (region_stack_.empty() || block_id == SemIR::InstBlockId::Unreachable) {
     return;
   }
 
-  if (!inst_block_stack().is_current_block_reachable()) {
-    // Don't include unreachable blocks in the function.
-    return;
+  region_stack_.back().block_ids.push_back(block_id);
+}
+
+auto Context::BeginSubpattern() -> void {
+  auto entry_block_id = sem_ir().inst_blocks().AddDefaultValue();
+  inst_block_stack().Push(entry_block_id);
+  PushRegion(entry_block_id);
```

Comment:
I'd find it more idiomatic to express this as
```suggestion
  inst_block_stack().Push();
  PushRegion(inst_block_stack().PeekOrAdd());
```